.github/workflows/nia-e2e.yaml

```
@@ -21,44 +21,88 @@ jobs:
+        channel: 1.28/edge/amc-connect
 
     - name: Tune Anbox Cloud
       run: |
         amc config set container.security_updates false
 
+    - name: Setup remote amc
+      env:
+        REMOTE_AMC_URL: ${{ vars.REMOTE_AMC_URL }}
+        REMOTE_CLIENT_CERT: ${{ secrets.REMOTE_CLIENT_CERT }}
+        REMOTE_CLIENT_KEY: ${{ secrets.REMOTE_CLIENT_KEY }}
+      run: |
+        set -x
+
+        url=${REMOTE_AMC_URL}
+        if [ -z "${url}" ]; then
+            ip=$(hostname -I | awk '{print $1}')
+            url="https://${ip}:8444"
+            amc config trust add "$HOME"/snap/anbox-cloud-appliance/current/client/client.crt
+        fi
+
+        if [ -n "${REMOTE_CLIENT_CERT}" ]; then
+          echo "${REMOTE_CLIENT_CERT}" >> "$HOME"/snap/anbox-cloud-appliance/current/client/remote_client.crt
+          amc config trust add "$HOME"/snap/anbox-cloud-appliance/current/client/remote_client.crt
+        fi
+
+        if [ -n "${REMOTE_CLIENT_KEY}" ]; then
+          echo "${REMOTE_CLIENT_KEY}" >> "$HOME"/snap/anbox-cloud-appliance/current/client/remote_client.key
+        fi
+
+        amc remote add remote_amc_client "${url}" --accept-certificate
+        amc remote set-default remote_amc_client
+
     - name: Restore cached images
       uses: actions/cache/restore@5a3ec84eff668545956fd18022155c47e93e2684 # v4
       with:
         path: images
         key: anbox-images-amd64
-
+        fail-on-cache-miss: true
+  
     - name: Import cached images
       run: |
         amc image add jammy:android${ANDROID_VERSION}:amd64 ./images/android${ANDROID_VERSION}.tar.xz
 
     - name: Create android instance
       id: create-instance
       run: |
-        id="$(amc launch -r -s adb jammy:android${ANDROID_VERSION}:amd64)"
+        sudo snap restart anbox-cloud-appliance.ams
+        sleep 30
+        id="$(amc launch --enable-streaming jammy:android${ANDROID_VERSION}:amd64)"
         amc wait -c status=running "$id"
-        echo "id=$id" >> "$GITHUB_OUTPUT"
+        echo "id=$id" >> "$GITHUB_OUTPUT"        
 
-    - name: Access android over adb
+    - name: Run e2e tests over adb
+      working-directory: ./nowinandroid
       run: |
+        set -x
         sudo apt install -y adb
-        id=${{ steps.create-instance.outputs.id }}
-        addr="$(amc show "$id" --format=json | jq -r .network.address)"
-        adb connect "$addr":5559
-        test "$(adb shell getprop ro.product.model)" = Anbox
 
-    - name: Run e2e tests
-      run: ./gradlew :app:connectedDemoDebugAndroidTest
-      working-directory: ./nowinandroid
+        id=${{ steps.create-instance.outputs.id }}
+        rm -fr logs && mkdir -p logs
+        amc connect -k "$id" |& tee logs/anbox-connect.log &
+        for i in {1..5}; do
+          grep "adb connect 127.0.0.1" logs/anbox-connect.log && break
+          sleep 5
+        done
+        port=$(grep -oP 'adb connect 127.0.0.1:\K\d+' logs/anbox-connect.log)
+                  
+        adb connect 127.0.0.1:"$port"
```

Nitpick: why not get both IP and port here?

Untested, but something like:
```suggestion
        url=$(grep -oP 'adb connect\s*\K.*' logs/anbox-connect.log)
                  
        adb connect "${url}"
```

I'm not sure of the exact format of amc connect's output, so this might not work as-is.

Wdyt?

---

sure i will try it but the original amc-connect already returns 127.0.0.1 with different port every run so i am not sure if this is necessary.